### PR TITLE
chore: bv_automata fails on simp-norm-form of constants

### PR DESCRIPTION
**Pull request by tobiasgrosser**

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
@AtticusKuhn could you have a look next week?

**Comment by AtticusKuhn** [comment]
Yes, I will look into this.

File: SSA/Experimental/Bits/Fast/Tactic.lean [hunk 1]
```diff
@@ -276,6 +294,19 @@ macro "bv_automata" : tactic =>
 # Test Cases
 -/
 
+def test_ok (x : BitVec 1) : 1 + x = x + 1 := by
```
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/Fast/Tactic.lean]: Maybe use more meaningful test names?

File: SSA/Experimental/Bits/Fast/Tactic.lean [hunk 2]
```diff
@@ -106,6 +106,18 @@ partial def first_rep (w : Q(Nat)) (e : Q( BitStream)) : SimpM (Σ (x : Q(BitStr
...
+      let length : Q(Nat) := w
+      let context  ← getLCtx
+      let contextLength := context.getFVarIds.size - 1
+      let lastFVar  ← context.getAt? contextLength
+      let qMapIndexToFVar : Q(Nat → BitStream) := .fvar lastFVar.fvarId
+      return ⟨
+        q(Term.eval (termNat $nat) $qMapIndexToFVar),
+        quoteThm qMapIndexToFVar length nat
+      ⟩
     | ~q(@BitStream.ofBitVec $w ($a - $b)) =>
```
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/Fast/Tactic.lean]:
```suggestion
    | ~q(BitStream.ofBitVec (BitVec.ofNat $w $b)) => do
      let .some nat := b.nat?
        | throwError m!"The bv_automata tactic expects {b} (representation form: {repr b}) to be of the form of a nat literal, but it is not"
      let length : Q(Nat) := w
      let context ← getLCtx
      let contextLength := context.getFVarIds.size - 1
      let lastFVar ← context.getAt? contextLength
      let qMapIndexToFVar : Q(Nat → BitStream) := .fvar lastFVar.fvarId
      return ⟨
        q(Term.eval (termNat $nat) $qMapIndexToFVar),
        quoteThm qMapIndexToFVar length nat
      ⟩
    | ~q(@BitStream.ofBitVec $w ($a - $b)) =>
```

**Comment by tobiasgrosser** [comment]
Otherwise, you can merge this. Thank you again.

**Pull request merged**